Rewrite the State Persistence demo. (#48)
DevExpress/devextreme-demos: JSDemos/Demos/DataGrid/StatePersistence/description.md

> **DataGrid** can persist its state. If a user sorts and filters data, groups, reorders and resizes columns, or makes other changes, the widget saves these modifications and restores them when a user reloads the page. Refer to the [stateStoring](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/) article for a full list of modifications that can be saved.

Reviewer: ```suggestion
The **DataGrid** can persist its state. If a user sorts and filters data, groups, reorders and resizes columns, or makes other changes, the widget saves these modifications and restores them when a user reloads the page. Refer to the [stateStoring](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/) article for a full list of modifications that can be saved.
```

> **DataGrid** state can be persistent - if a user sorts and filters data, groups, reorders and resizes columns, or makes other changes, the widget saves these modifications and restores them when a user reloads the page. Refer to the [stateStoring](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/) article for a full list of modifications that can be saved.

Reviewer 2: I'd change "state can be persistent" to "can persist its state". Better use a strong verb instead of any version of "to be".

 ```suggestion
**DataGrid** can persist its state. If a user sorts and filters data, groups, reorders and resizes columns, or makes other changes, the widget saves these modifications and restores them when a user reloads the page. Refer to the [stateStoring](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/) article for a full list of modifications that can be saved.
```

> **DataGrid** state can be persistent - if a user sorts and filters data, groups, reorders and resizes columns or makes other changes, then the widget saves these modifications and restores them when a user reloads the page. The full list of modifications that can be saved is in the [stateStoring](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/) documentation.

Reviewer 3: - "... or makes other changes..." - Oxford comma missing
- "... then the widget saves these modifications..." - "then" associates with time or order. Neither is the case.
- "The full list of modifications **that can be saved**" - unnecessary adjective clause
- "The full list of modifications ... is in the stateStoring documentation" - "is" is a weak verb. Actually, I've rearranged the last sentence because, to my mind, imperative mood fits better here.

```suggestion
**DataGrid** state can be persistent - if a user sorts and filters data, groups, reorders and resizes columns, or makes other changes, the widget saves these modifications and restores them when a user reloads the page. Refer to the [stateStoring](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/) article for a full list of saved modifications.
```

Author: @RomanTsukanov 
* "... then the widget saves these modifications..." - "then" grammatically fits into this sentence (see https://forum.thefreedictionary.com/postst156297_if---then.aspx). This word can be ommited though, but I used it to stress the beginning of the main clause. Should it be here?

* "saved modifications" -> "savable modifications": "savable" suits better, doesn't it?

* Is stateStoring an article or a documentation?

Reviewer 3: - I see why you used "then", but it is unnecessary.
- "Savable" is a pretty rare word, better play safe. Let's return the adjective clause.
- Documentation is a whole piece. It consists of various articles: tutorials, guides, etc.

```suggestion
**DataGrid** state can be persistent - if a user sorts and filters data, groups, reorders and resizes columns, or makes other changes, the widget saves these modifications and restores them when a user reloads the page. Refer to the [stateStoring](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/) article for a full list of modifications that can be saved.
```

> To enable state persistence, set **stateStoring**.[enabled](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#enabled) to **true** and specify the [storageKey](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#storageKey) and [type](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#type) options. Depending on the type, the state can be saved to a [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) or [sessionStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage). With localStorage, the state persists across browser sessions; with sessionStorage, it is reset after the current session. This demo uses localStorage.

Reviewer 4: When you refer to the third-party resources, use HTML links instead of markdown. This rule exists because markdown does not allow you to open a link in a new tab, and the latter is a required condition. So it should be 
`<a href="https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage" target="_blank">localStorage</a>`. The same applies to the sessionStorage link.

I would also make these names bold when they are not links.

Author: Thank you. Applied these edits.

> To enable state persistence, set **stateStoring**.[enabled](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#enabled) to **true** and specify the [storageKey](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#storageKey) and [type](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#type) options. Depending on the type, the state can be saved to a <a href="https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage" target="_blank">localStorage</a> or <a href="https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage" target="_blank">sessionStorage</a>. With **localStorage**, the state persists across browser sessions; with **sessionStorage**, it is reset after the current session. This demo uses **localStorage**.

Reviewer 3: ```suggestion
To enable state persistence, set **stateStoring**.[enabled](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#enabled) to **true** and specify the [storageKey](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#storageKey) and [type](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#type) options. Depending on the **type**, the state can be saved to a <a href="https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage" target="_blank">localStorage</a> or <a href="https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage" target="_blank">sessionStorage</a>. With a **localStorage**, the state persists across browser sessions; with a **sessionStorage**, it is reset after the current session. This demo uses a **localStorage**.
```

> Alternatively, you can create a custom storage and implement the [customSave](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customSave) and [customLoad](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customLoad) functions if you need to store data in accordance with a custom logic.

Reviewer 3: "you can create a custom storage" - I have two remarks regarding this piece:
1) I would rather suggest that the developer already has a custom storage. Then, this phrase should read "if you want to save the state in a custom storage", but (see 2)
2) customSave and customLoad can be used without a custom storage, for example, to tweak the state a bit before saving it to localStorage. So, I think the following version would be more accurate:

```suggestion
Alternatively, you can implement the [customSave](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customSave) and [customLoad](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customLoad) functions to save and load the state according to a custom logic.
```

Author: @RomanTsukanov 
"create a custom storage" is erroneous here.
Since "customSave and customLoad can be used without a custom storage", should there be a sentence that it is possible to **use** a custom storage (this implies a change of the stateStoring's type) and **implement** these two functions?

Reviewer 3: ```suggestion
Alternatively, you can implement the [customSave](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customSave) and [customLoad](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customLoad) functions to save and load the state according to a custom logic. In this case, set the **type** option to *"custom"*.
```

> Alternatively, you can implement the [customSave](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customSave) and [customLoad](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customLoad) functions to save and load the state according to a custom logic. In this case, set the **type** option to *"custom"*.

Reviewer: ```suggestion
Alternatively, you can implement the [customSave](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customSave) and [customLoad](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/#customLoad) functions to save and load the state according to custom logic. In this case, set the **type** option to *"custom"*.
```
